tests/queries/tests.py

```diff
@@ -2104,6 +2104,20 @@ def test_sliced_delete(self):
         DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id')[1:]).delete()
         self.assertEqual(set(DumbCategory.objects.values_list('id', flat=True)), {3})
 
+    def test_ticket_25811(self):
+        # Regular checking with a subquery
+        query = DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id')[2:])
+        self.assertEqual(set(query.values_list('id', flat=True)), {1, 2})
+
+        # Regular checking with a list
+        query = DumbCategory.objects.filter(id__in=[1, 2])
+        self.assertEqual(set(query.values_list('id', flat=True)), {1, 2})
+
+        # Error Checking subquery to generate a different database
+        with self.assertRaises(ValueError):
+            query = DumbCategory.objects.filter(id__in=DumbCategory.objects.using('other').order_by('-id')[2:])
+            self.assertEqual(set(query.values_list('id', flat=True)), {1, 2})
```

Comment:
I assume the whole `assertEqual()` is copied from above, but it is just noise here. All you want to do is force the query to be evaluated, so use `list(query)`.